Swiftification/URL.swift

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright (c) 2019 Hilton Campbell
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+import Foundation
+
+public extension URL {
+    
+    private func fallbackURLByAppendingQueryString(parameters: [String: String]) -> URL {
+        guard !parameters.isEmpty, let scheme = scheme, let host = host else { return self }
+        let urlString = scheme + "://" + host + path + "?" + parameters.map { "\($0)=\($1)" }.sorted().joined(separator: "&")
```

Comment:
I'm nervous about having this fallback because it doesn't do any escaping and there could be other edge cases that aren't handled. I'd prefer to return a `URL?` from the other method.